Update train.py

chris-ha458 opened the pull request:
update train.py 
1. black formatter.
2. remove unnecessary import
3. add more arguments

satpalsr committed:
add flash_attn_kvpacked

satpalsr committed:
fix formatting

dashstander committed:
whoops

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

satpalsr committed:
accept changes from main & resolve conflicts

satpalsr committed:
Merge branch 'main' into flash_attn_infer

dashstander committed:
Merge in new changes

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

dashstander committed:
Error

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

dashstander committed:
errors

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

SauravMaheshkar committed:
feat(ci): add pip caching to CI

dashstander committed:
Set training attribute appropriately

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

dashstander committed:
Split up FlashAttention methods

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

dashstander committed:
Comment out clear_cache

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

dashstander committed:
Just remove clear_cache

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

dashstander committed:
Fix pre-commit formatting

Signed-off-by: Dashiell Stander <dstander@protonmail.com>

StellaAthena committed:
Merge pull request #889 from SauravMaheshkar/main

feat(ci): add `pip` caching to CI

StellaAthena committed:
Merge branch 'main' into flash_attn_infer

StellaAthena committed:
Merge pull request #862 from satpalsr/flash_attn_infer

add flash_attn_kvpacked

curt-tigges committed:
Changed is_pipe_parallel setting to fix pipeline-parallel inference (#866)

* Changed is_pipe_parallel setting to fix pipeline-parallel inference

* Update NeoXArgs docs automatically

* Update NeoXArgs docs automatically

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Quentin Anthony <qganthony@yahoo.com>

SauravMaheshkar committed:
feat: improve typing

StellaAthena committed:
Merge pull request #893 from SauravMaheshkar/main

feat(logging): improve typing

StellaAthena committed:
Added DeeperSpeed to requirements.txt

Commit:
Update NeoXArgs docs automatically

Commit:
Update NeoXArgs docs automatically

StellaAthena committed:
Merge pull request #895 from EleutherAI/reqs-correction

Reqs correction

chris-ha458 committed:
Update train.py

update train.py 
1. black formatter.
2. remove unnecessary import
3. add more arguments

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/gpt-neox?pullRequest=897) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/gpt-neox?pullRequest=897) before we can accept your contribution.<br/>**4** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: satpalsr<br/>:white_check_mark: dashstander<br/>:white_check_mark: StellaAthena<br/>:white_check_mark: chris-ha458<br/>:x: SauravMaheshkar<br/>:x: curt-tigges<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/gpt-neox?pullRequest=897) it.</sub>

chris-ha458 committed:
Update utils.py

Black formatting
Add logic required to expand "~"

chris-ha458 committed:
Update train.py

removed num_proc
temporarily disabled emoji
added continuing subword prefix option ( does not work well with Bytelevel)

chris-ha458 committed:
Update utils.py

improve reader error handling

chris-ha458 committed:
Update train.py

add whitespace related handling.
add whitespace argument expose
reconstruct pre_tokenizer_list
add more whitespace to check tokenizer invertibility

chris-ha458 committed:
Update train.py

chris-ha458 committed:
Update utils.py

remove unnecessary print

chris-ha458 committed:
Update train.py

set dropout default to None
import path related code.
Change normalizer
change buffer_tokens
change whitespace reservation handling

chris-ha458 committed:
Update train.py

Clear whitespace_reservation TODO
add single_whitespace argument (might be necessary for invertibility)

chris-ha458 committed:
Create .gitignore

add gitignore file to ignore artifacts

chris-ha458 committed:
Update train.py

add directory parsing error checks
add more metrics
(tokenizer reconstructions, unicode fallback portion)

chris-ha458 committed:
Update preprocess.py

path handling changes
black formatting

chris-ha458 committed:
Update train.py

change from GPT2TokenizerFast to PreTrainedTokenizerFast class

chris-ha458 committed:
Update train.py

enhanced test string

chris-ha458 committed:
Update utils.py

add logic to handle jsonl, txt input
add logic to handle folder with jsonl,txt or arrow dataset

chris-ha458 committed:
Update train.py

add byte_fallback option expose
(incompatible with current transformer wrapper)
change dataset_loading with new util.py
add dataset shuffling option

chris-ha458 committed:
Update utils.py

fix error in loading sequence

chris-ha458 committed:
Update train.py

fix whitespace preservation logic

chris-ha458 committed:
Update train.py

simplify data loading logic.
remove unnecessary special tokens

chris-ha458 committed:
Update train.py

remove emoji related code

chris-ha458 committed:
Update train.py

add whitespace processing regex
r"\s{16,}"

chris-ha458 committed:
update tokenizer

add whitespace pretokenizer
(only processes looong whitespaces)

chris-ha458 committed:
Merge branch 'main' into patch-1

chris-ha458 committed:
Update train.py

chris-ha458 committed:
Update train.py

add camel case regex

chris-ha458 committed:
Update train.py

separate camel_case regex

chris-ha458 committed:
Update train.py

chris-ha458 committed:
Update train.py

bzantium commented:
@satpalsr @dashstander @StellaAthena Is it intentional to add commits on this branch or PR?

dashstander commented:
I have not myself touched this PR

bzantium commented:
> I have not myself touched this PR

Oh I think this happened while merging other updates. Thanks for clarify :)
